doc/deployment/amazon_web_services/aws-deploy-pachyderm.md

```diff
@@ -0,0 +1,238 @@
+If you created your cluster by using `kops` or `eksctl`
+the nodes must have a dedicated IAM role already assigned.
+
+* The IAM role must have access to the S3 bucket that you created for
+Pachyderm.
+
+* The IAM role must have correct trust relationships.
+
+To deploy Pachyderm with an IAM role, complete the following steps:
+
+1. Find the IAM role assigned to the cluster:
+
+   1. Go to the AWS Management console.
+   1. Select an EC2 instance in the Kubernetes cluster.
+   1. Click **Description**.
+   1. Find the **IAM Role** field.
+
+1. Enable access to the S3 bucket for the IAM role:
+
+   1. In the **IAM Role** field, click on the IAM role.
+   1. In the **Permissions** tab, click **Edit policy**.
+   1. Select the **JSON** tab.
+   1. Append the following text to the end of the existing JSON:
+
+      ```json
+      {
+          "Effect": "Allow",
+              "Action": [
+                  "s3:ListBucket"
+              ],
+              "Resource": [
+                  "arn:aws:s3:::<your-bucket>"
+              ]
+      },
+      {
+          "Effect": "Allow",
+          "Action": [
+              "s3:PutObject",
+          "s3:GetObject",
+          "s3:DeleteObject"
+          ],
+          "Resource": [
+              "arn:aws:s3:::<your-bucket>/*"
+          ]
+      }
+      ```
+
+      Replace `<your-bucket>` with the name of your S3 bucket.
+
+      **Note:** For the EKS cluster, you might need to use the
+      **Add inline policy** button and create a name for this policy.
+
+1. Set up trust relationships for the IAM role:
+
+   1. Click the **Trust relationships > Edit trust relationship**.
+   1. Ensure that you see a statement with `sts:AssumeRole`. Example:
+
+      ```json
+      {
+        "Version": "2012-10-17",
+        "Statement": [
+          {
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "ec2.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+          }
+        ]
+      }
+      ```
+
+1. Deploy Pachyderm:
+
+   ```bash
+   $ pachctl deploy amazon ${BUCKET_NAME} ${AWS_REGION} ${STORAGE_SIZE} --dynamic-etcd-nodes=1 --iam-role <your-iam-role>
+   ```
+
+   The deployment takes some time. You can run `kubectl get pods` periodically
+   to check the status of deployment. When Pachyderm is deployed, the command
+   shows all pods as `READY`:
+
+   ```bash
+   $ kubectl get pods
+   NAME                     READY     STATUS    RESTARTS   AGE
+   dash-6c9dc97d9c-89dv9    2/2       Running   0          1m
+   etcd-0                   1/1       Running   0          4m
+   pachd-65fd68d6d4-8vjq7   1/1       Running   0          4m
+   ```
+
+   **Note:** If you see a few restarts on the `pachd` nodes, that means that
+   Kubernetes tried to bring up those pods before `etcd` was ready. Therefore,
+   Kubernetes restarted those pods. You can safely ignore this message.
+
+1. In a new terminal window, forward Pachyderm ports to enable access to
```

Comment:
In general, we're moving away from port-forward and should stop recommending it within the docs. Here specifically, implicit port forward is sufficient and no this can be removed entirely.